src/tool/cppxlang/strings/base_string.h

```diff
@@ -345,70 +322,70 @@ namespace winrt
     inline hstring to_hstring(uint8_t value)
     {
         wchar_t buffer[32];
-        swprintf_s(buffer, L"%hhu", value);
+        swprintf(buffer, 32, L"%hhu", value);
```

Comment:
Consider `countof()` or `ARRAYSIZE()` or similar instead of repeating the literal size.

Reply:
`_countof` is VC-specific. Perhaps `std::array` to avoid repeating 32.